manifests/ssltools/nssdb.pp

```diff
@@ -1,50 +1,26 @@
 # Sets up nssdb
 class certs::ssltools::nssdb (
-  $nss_db_dir = $certs::nss_db_dir,
+  $nss_db_dir = "${certs::pki_dir}/nssdb",
+  $nss_db_password_file = "${certs::pki_dir}/nss_db_password-file",
+  $nss_db_password = extlib::cache_data('foreman_cache_data', 'certs-nss-db-password', extlib::random_password(32)),
   $group = 'qpidd',
```

Comment:
Just a last thing: can we use data types?
```suggestion
  Stdlib::Absolutepath $nss_db_dir = "${certs::pki_dir}/nssdb",
  Stdlib::Absolutepath $nss_db_password_file = "${certs::pki_dir}/nss_db_password-file",
  String[10] $nss_db_password = extlib::cache_data('foreman_cache_data', 'certs-nss-db-password', extlib::random_password(32)),
  String[1] $group = 'qpidd',
```

Reply:
That's a good call out.